vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -308,8 +311,15 @@ public void setLocale(Locale locale) {
     }
 
     private void setLocaleWithJS() {
```

Comment:
We should extend the `setLocale` API docs to explain that setting a locale has no effect when using custom date formats.

We can also consider logging a warning in dev mode.

Reply:
Added note to setLocale javadoc